src/client/driver/driver.js

```diff
@@ -339,18 +411,37 @@ export default class Driver {
         // but it should wait until the page is loaded before executing a command.
         this.readyPromise
             .then(() => {
+                if (this.dialogsMonitor && !isHandleDialogCommand(command))
+                    this.dialogsMonitor.expectedDialogs = command.expectedDialogs;
+
+                var isCommandRejectableByError = isCommandRejectableByPageError(command);
+                var pendingPageError           = this.contextStorage.getItem(PENDING_PAGE_ERROR);
+                var needFailWithPageError      = pendingPageError && isCommandRejectableByError;
+
+                if (needFailWithPageError) {
+                    this._onReady(new DriverStatus({ isCommandResult: true }));
+                    return;
+                }
+
+                var isThereActiveIframe = this.activeChildDriverLink ||
+                                          this.contextStorage.getItem(ACTIVE_IFRAME_SELECTOR);
+
+                var needExecuteInIframe = !isExecutableInTopWindowOnly(command) && isThereActiveIframe;
```

Comment:
`isExecutableInTopWindowOnly` and `isThereActiveIframe` are quite clear themselves